R/conditions.R

```diff
@@ -127,6 +127,49 @@ stop_incompatible_op <- function(op, x, y, details = NULL, ..., message = NULL,
   )
 }
 
+#' @rdname vctrs-conditions
+#' @export
+stop_incompatible_size <- function(x, y,
+                                   x_size, y_size,
+                                   x_arg = "", y_arg = "",
+                                   details = NULL,
+                                   ...,
+                                   message = NULL,
+                                   .subclass = NULL) {
+  vec_assert(x)
+  vec_assert(y)
+
+  if (is_null(message)) {
+    if (nzchar(x_arg)) {
+      x_name <- paste0("`", x_arg, "` of size")
+    } else {
+      x_name <- "vector of size"
+    }
+    if (nzchar(y_arg)) {
+      y_name <- paste0("`", y_arg, "` of size")
+    } else {
+      y_name <- "vector of size"
+    }
+
+    message <- glue_lines(
+      "No common size for {x_name} [{x_size}] and {y_name} [{y_size}].",
```

Comment:
Would it be more with `stop_incompatible_type()` to generate something like ``No common size for `x`[3] and `y`[4]`` ? Otherwise I don't think you'd want the square brackets, and generate ``No common size for `x`, size 3, and `y`, size 4.``

Reply:
I like the second. What should be the message when no arg is passed?

```
No common size for vector, size 3, and vector, size 4.
```

or

```
No common size for vector of size 3, and vector of size 4.
```